verification/src/compiler/version.rs

```diff
@@ -0,0 +1,115 @@
+use semver::Version;
+use std::{fmt::Display, str::FromStr};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ReleaseVersion {
+    pub version: Version,
+    pub commit: String,
+}
+
+impl ReleaseVersion {
+    pub fn new(version: Version, commit: String) -> Self {
+        Self { version, commit }
+    }
+}
+
+/// Parses release version from string formated as
+/// `solc-v*VERSION*+commit.*COMMITHASH*`, example
+/// `solc-v0.8.9+commit.e5eed63a`
+impl FromStr for ReleaseVersion {
+    type Err = anyhow::Error;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parsed = sscanf::scanf!(s, "solc-v{String}+commit.{String}")
+            .map_err(|e| anyhow::anyhow!("{:?}", e))?;
+        let version = Version::from_str(&parsed.0)?;
+        let commit = parsed.1;
+        if commit.len() != 8 {
+            anyhow::bail!("expected commit hash of length 8, got {}", commit);
+        }
+        Ok(Self { version, commit })
+    }
+}
+
+impl Display for ReleaseVersion {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "solc-v{}+commit.{}", self.version, self.commit)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct NightlyVersion {
+    pub version: Version,
+    pub date: String,
```

Comment:
Consider using `chrono::NaiveDate` and `chrono::NaiveDate::parse_from_str` to parse the string, so to ensure that the date string is also valid

Reply:
we're really don't care much about date, while we could match by commit hash, date is mostly metainformation imo. I store it mostly so display will output the same string

Reply:
(also it probably will make some trouble because chrono accepts "2000-01-30" and we have "2000.1.30"

Reply:
> we're really don't care much about date, while we could match by commit hash, date is mostly metainformation imo. I store it mostly so display will output the same string

Get it. Then yes, probably we may omit checking the date here, though I personally still believe that it would be nice to have feature.

> (also it probably will make some trouble because chrono accepts "2000-01-30" and we have "2000.1.30")
Seems that we may parse with `chrono::NaiveDate::parse_from_str(s,  "%Y.%m.%d")`